src/mscorlib/src/System/Lazy.cs

```diff
@@ -322,147 +516,12 @@ public T Value
         {
             get
             {
-                Boxed boxed = null;
-                if (m_boxed != null )
-                {
-                    // Do a quick check up front for the fast path.
-                    boxed = m_boxed as Boxed;
-                    if (boxed != null)
-                    {
-                        return boxed.m_value;
-                    }
-
-                    LazyInternalExceptionHolder exc = m_boxed as LazyInternalExceptionHolder;
-                    Debug.Assert(exc != null);
-                    exc.m_edi.Throw();
-                }
-
-                // Fall through to the slow path.
-                return LazyInitValue();
+                var state = _state;
+                if (state == null)
+                    return _value;
+                return LazyGetValue(state);
```

Comment:
We want to optimize for the initialized path.  Would the compiled code be smaller/fast/more-inlineable/etc. if this were instead written as:
```C#
public T Value => _state == null ? _value : LazyGetValue();
```
and have LazyGetValue access _state again?  I realize it'd be another volatile read in the non-initialized path, but in that path we're going to do a whole lot of other work as well, e.g. execute a delegate.